contracts/facets/EmbalmerFacet.sol

```diff
@@ -64,32 +96,51 @@ contract EmbalmerFacet {
                 cursedBondAmount
             );
 
-            // Add to the neccessary data structures
+            // If the archaeologist is the arweave archaeologist, set the
+            // storage fee. This is the only storage fee we care about.
+            if (archaeologists[i].archAddress == arweaveArchaeologist) {
+                storageFee = archaeologists[i].storageFee;
+            }
```

Comment:
Does there need to be any validation that the arweaveArchaeologist address *is* present in one of the archeologist structs? Otherwise this storage fee would never get set.

alternatively, it could validate the storage fee is not 0 outside of this loop, which means this conditional returned true at some point.

Reply:
Very good point. I think we should go with the latter option to avoid iterating over the list of archaeologists.